docs/api/finder.rst

```diff
@@ -0,0 +1,12 @@
+.. _kernel_finder_api:
+
+Kernel Finder
+=============
+
+The Kernel Finder API is used by applications wishing to discover, launch,
+and manage Jupyter kernels.  There is no facility to *extend* class ``KernelFinder``.
```

Comment:
This is a bit unclear - installing additional providers is kind of extending KernelFinder. Maybe say something like "this is not meant to be subclassed - to make it find additional kernels, see *Kernel Providers*"